tests/feedback.test.ts

```diff
@@ -0,0 +1,195 @@
+import { validate } from 'class-validator';
+import { plainToClass } from 'class-transformer';
+import { DatabaseConnection, PortalState, UserFactory } from './data';
+import { FeedbackFactory } from './data/FeedbackFactory';
+import { ActivityScope, ActivityType, FeedbackStatus, UserAccessType } from '../types';
+import { ActivityModel } from '../models/ActivityModel';
+import { FeedbackModel } from '../models/FeedbackModel';
+import { Feedback } from '../api/validators/FeedbackControllerRequests';
+import { UserModel } from '../models/UserModel';
+import { Config } from '../config';
+import { ControllerFactory } from './controllers';
+
+beforeAll(async () => {
+  await DatabaseConnection.connect();
+});
+
+beforeEach(async () => {
+  await DatabaseConnection.clear();
+});
+
+afterAll(async () => {
+  await DatabaseConnection.clear();
+  await DatabaseConnection.close();
+});
+
+describe('feedback submission', () => {
+  test('properly persists on successful submission', async () => {
+    const conn = await DatabaseConnection.get();
+    const [user] = UserFactory.create(1);
+    const [feedback] = FeedbackFactory.create(1);
+
+    const feedbackController = ControllerFactory.feedback(conn);
+    await new PortalState().createUsers([user]).write(conn);
+
+    await ControllerFactory.feedback(conn).submitFeedback({ feedback }, user);
+    const submittedFeedbackResponse = await feedbackController.getFeedback(user);
+
+    expect(submittedFeedbackResponse.feedback).toHaveLength(1);
+    expect(submittedFeedbackResponse.feedback[0]).toStrictEqual({
+      ...submittedFeedbackResponse.feedback[0],
+      user: user.getPublicProfile(),
+      status: FeedbackStatus.SUBMITTED,
+      ...feedback,
+    });
+  });
+
+  test('is invalidated when submission description is too short', async () => {
+    const [feedback] = FeedbackFactory.with({ description: 'A short description' });
+
+    const errors = await validate(plainToClass(Feedback, feedback));
+
+    expect(errors).toBeDefined();
+    expect(errors).toHaveLength(1);
+    expect(errors[0].property).toEqual('description');
+    expect(errors[0].constraints.minLength).toBeDefined();
+  });
+
+  test('persists submission data and activity in database on successful submission', async () => {
```

Comment:
Doesn't the sample test already verify that things get persisted to the database?

Reply:
Yup - I think what we want to assert instead is that the scope/type are correct for feedback submission

Reply:
Also, slightly unrelated, do we want `PublicActivity` to return the `scope` as well? Since `PublicActivity` can now be either private or public depending on if it's fetched for the current user or a different one. Also we should probably rename `PublicActivity` since it's technically incorrect but still conventional since everything returned to the client is considered "public"

Reply:
I don't think `PublicActivity` needs to return `scope` since permissions are internal, but I don't have a strong opinion about it. The type name's probably fine to leave as is.